lnrpc/routerrpc/router_backend.go

```diff
@@ -343,7 +348,12 @@ func (r *RouterBackend) QueryRoutes(ctx context.Context,
 	// Calculate route success probability. Do not rely on a probability
 	// that could have been returned from path finding, because mission
 	// control may have been disabled in the provided ProbabilitySource.
-	successProb := r.getSuccessProbability(route)
+	successProb, err := r.getSuccessProbability(
```

Comment:
Still wondering if with the newly added extra code and lookup it isn't better to return the probably along with the route from pathfinding. Pathfinding already has it.

Reply:
I have added a commit that returns the probability from `findPath` to be usable in `QueryRoutes`. Advantage is that we don't have to recompute the probability (see fixup commit that removes `getSuccessProbability`), but the downside is that we don't get probability estimates anymore if `use_mc` is disabled. Unfortunately we can't remove the capacity lookup code because `QueryProbability` still needs it.

Reply:
I think it's a good change. Probabilities without `use_mc` doesn't make that much sense anyway.